### PR TITLE
CIP-1694 CLI stage 2

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

This represents the MVP for [CIP-1694](https://github.com/cardano-foundation/CIPs/blob/e96681908fd336702b9d89827063a190c5656eab/CIP-1694/README.md)

# Changelog

```yaml
- description: |
    <insert-changelog-description-here>
  # no-changes: the API has not changed
  # compatible: the API has changed but is non-breaking
  # breaking: the API has changed in a breaking way
  compatibility: <no-api-changes|compatible|breaking>
  # feature: the change implements a new feature in the API
  # bugfix: the change fixes a bug in the API
  # test: the change fixes modifies tests
  # maintenance: the change involves something other than the API
  # If more than one is applicable, it may be put into a list.
  type: <feature|bugfix|test|maintenance>
```

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG node developers to do this
for you.

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -41,3 +41,10 @@ write-ghc-environment-files: always
 -- Do NOT add more source-repository-package stanzas here unless they are strictly
 -- temporary! Please read the section in CONTRIBUTING about updating dependencies.
 
+source-repository-package
+  type: git
+  location: https://github.com/input-output-hk/cardano-api
+  tag: 2010540079b53c72376b69dd72216ef1260150d0
```
Comment (andreabedini) [inline cabal.project]: This commit does not seem to exist?

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 2]
```diff
@@ -69,6 +73,192 @@ Usage: cardano-cli governance create-genesis-key-delegation-certificate
...
+Usage: cardano-cli governance committee key-gen-cold 
+            --cold-verification-key-file FILE
+            --cold-signing-key-file FILE
+
+  Create a cold key pair for a Constitutional Committee Memberkey and a new
+  certificate issue counter
+
+Usage: cardano-cli governance committee key-gen-hot --verification-key-file FILE
+            --signing-key-file FILE
+            --operational-certificate-issue-counter-file FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli:176]: Please delete  `--operational-certificate-issue-counter-file FILE `

We need no operational certificate counter file.

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 3]
```diff
@@ -69,6 +73,192 @@ Usage: cardano-cli governance create-genesis-key-delegation-certificate
...
+            ( --drep-verification-key STRING
+            | --drep-cold-verification-key-file FILE
+            )
+            [--metadata-url URL --metadata-hash HASH]
+            --out-file FILE
+
+  Create a DRep registration certificate
+
+Usage: cardano-cli governance drep deregistration-certificate 
+            ( --drep-verification-key STRING
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli:226]: It should be `--drep-cold-verification-key STRING`

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 4]
```diff
@@ -69,6 +73,192 @@ Usage: cardano-cli governance create-genesis-key-delegation-certificate
...
+            | --drep-cold-verification-key-file FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli:227]: Dreps do not use cold keys. it should be `--drep-verification-key-file FILE`

File: cardano-cli/test/cardano-cli-golden/files/golden/help.cli [hunk 5]
```diff
@@ -69,6 +73,192 @@ Usage: cardano-cli governance create-genesis-key-delegation-certificate
...
+            )
+            --epoch NATURAL
+            --out-file FILE
+
+  Create a DRep deregistration certificate
+
+Usage: cardano-cli governance drep id 
+            ( --drep-verification-key STRING
+            | --drep-cold-verification-key-file FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help.cli:236]: should be  `--drep-verification-key-file`

File: cardano-cli/test/cardano-cli-golden/files/golden/help/governance_action_create-constitution-update.cli [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Usage: cardano-cli governance action create-constitution-update --out-file FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/governance_action_create-constitution-update.cli:1]: What is the content of the out-file this way??

I think it should be at minimum something like:

```
cardano-cli governance action create-constitution-update
--proposal-hash STRING
--out-file
```
where we can specify what algorithm to use or ideally provide a tool on the cli. 

I know it mostly depends on ledger but I have already discussed with Alexey and Jared. User should provide a URL to the document proposed and its hash, and ideally a compressed version of the proposal as auxiliary data. this way the compressed version will always be ON-CHAIN 

```
cardano-cli governance action create-constitution-update
--proposal-url  URL
--proposal-hash STRING
--proposal-compressed-file FILE? RAW-BYTES?
--out-file
```

File: cardano-cli/test/cardano-cli-golden/files/golden/help/governance_action_create-treasury-withdrawal.cli [hunk 7]
```diff
@@ -0,0 +1,7 @@
+Usage: cardano-cli governance action create-treasury-withdrawal --out-file FILE
```
Comment (CarlosLopezDeLara) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/governance_action_create-treasury-withdrawal.cli:1]: I think it should be something like: 

```
cardano-cli governance action create-treasury-withdrawal       
(--stake-address ADDRESS --lovelace LOVELACE)
--metadata-url 
--metadata-hash
--out-file
```

**Pull request closed without merging**